docs/src/modules/components/AppNavDrawer.js

```diff
@@ -456,14 +456,14 @@ export default function AppNavDrawer(props) {
               versionSelector={renderVersionSelector([
```

Comment:
This is also a good opportunity to fix missing product identifier on some of the MUI X pages like https://mui.com/x/whats-new/

<img width="472" alt="Screenshot 2023-03-03 at 23 26 52" src="https://user-images.githubusercontent.com/13808724/222842561-84c3e747-c72c-40f9-a54b-1592e6edb418.png">

It would make way more sense to handle these `if` statements as mutually exclusive:

```diff
diff --git a/docs/src/modules/components/AppNavDrawer.js b/docs/src/modules/components/AppNavDrawer.js
index 5f997aa9f3..4fb05fbf17 100644
--- a/docs/src/modules/components/AppNavDrawer.js
+++ b/docs/src/modules/components/AppNavDrawer.js
@@ -277,6 +277,89 @@ function reduceChildRoutes(context) {
   return items;
 }
 
+function XProductIdentifier({ pathname, languagePrefix, renderVersionSelector }) {
+  if (!pathname.startsWith('/x/')) {
+    return null;
+  }
+  if (pathname.startsWith('/x/react-data-grid/') || pathname.startsWith('/x/api/data-grid/')) {
+    return (
+      <ProductIdentifier
+        name="Data Grid"
+        metadata="MUI X"
+        versionSelector={renderVersionSelector([
+          // DATA_GRID_VERSION is set from the X repo
+          {
+            text: 'v6',
+            ...(process.env.DATA_GRID_VERSION.startsWith('6')
+              ? {
+                  text: `v${process.env.DATA_GRID_VERSION}`,
+                  current: true,
+                }
+              : {
+                  href: `https://mui.com${languagePrefix}/components/data-grid/`,
+                }),
+          },
+          {
+            text: 'v5',
+            ...(process.env.DATA_GRID_VERSION.startsWith('5')
+              ? {
+                  text: `v${process.env.DATA_GRID_VERSION}`,
+                  current: true,
+                }
+              : {
+                  href: `https://v5.mui.com${languagePrefix}/components/data-grid/`,
+                }),
+          },
+          { text: 'v4', href: `https://v4.mui.com${languagePrefix}/components/data-grid/` },
+        ])}
+      />
+    );
+  }
+  if (
+    pathname.startsWith('/x/react-date-pickers/') ||
+    pathname.startsWith('/x/api/date-pickers/')
+  ) {
+    return (
+      <ProductIdentifier
+        name="Date pickers"
+        metadata="MUI X"
+        versionSelector={renderVersionSelector([
+          // DATE_PICKERS_VERSION is set from the X repo
+          {
+            ...(process.env.DATE_PICKERS_VERSION.startsWith('6')
+              ? {
+                  text: `v${process.env.DATE_PICKERS_VERSION}`,
+                  current: true,
+                }
+              : {
+                  text: `v6-next`,
+                  href: `https://next.mui.com${languagePrefix}/components/data-grid/`,
+                }),
+          },
+          {
+            ...(process.env.DATE_PICKERS_VERSION.startsWith('5')
+              ? {
+                  text: `v${process.env.DATE_PICKERS_VERSION}`,
+                  current: true,
+                }
+              : {
+                  text: `v5`,
+                  href: `https://mui.com${languagePrefix}/components/data-grid/`,
+                }),
+          },
+        ])}
+      />
+    );
+  }
+  return <ProductIdentifier name="Advanced components" metadata="MUI X" />;
+}
+
+XProductIdentifier.propTypes = {
+  languagePrefix: PropTypes.string.isRequired,
+  pathname: PropTypes.string.isRequired,
+  renderVersionSelector: PropTypes.func.isRequired,
+};
+
 // iOS is hosted on high-end devices. We can enable the backdrop transition without
 // dropping frames. The performance will be good enough.
 // So: <SwipeableDrawer disableBackdropTransition={false} />
@@ -445,74 +528,11 @@ export default function AppNavDrawer(props) {
               ])}
             />
           )}
-          {canonicalAs.startsWith('/x/introduction/') && (
-            <ProductIdentifier name="Advanced components" metadata="MUI X" />
-          )}
-          {(canonicalAs.startsWith('/x/react-data-grid/') ||
-            canonicalAs.startsWith('/x/api/data-grid/')) && (
-            <ProductIdentifier
-              name="Data Grid"
-              metadata="MUI X"
-              versionSelector={renderVersionSelector([
-                // DATA_GRID_VERSION is set from the X repo
-                {
-                  text: 'v6',
-                  ...(process.env.DATA_GRID_VERSION.startsWith('6')
-                    ? {
-                        text: `v${process.env.DATA_GRID_VERSION}`,
-                        current: true,
-                      }
-                    : {
-                        href: `https://mui.com${languagePrefix}/components/data-grid/`,
-                      }),
-                },
-                {
-                  text: 'v5',
-                  ...(process.env.DATA_GRID_VERSION.startsWith('5')
-                    ? {
-                        text: `v${process.env.DATA_GRID_VERSION}`,
-                        current: true,
-                      }
-                    : {
-                        href: `https://v5.mui.com${languagePrefix}/components/data-grid/`,
-                      }),
-                },
-                { text: 'v4', href: `https://v4.mui.com${languagePrefix}/components/data-grid/` },
-              ])}
-            />
-          )}
-          {(canonicalAs.startsWith('/x/react-date-pickers/') ||
-            canonicalAs.startsWith('/x/api/date-pickers/')) && (
-            <ProductIdentifier
-              name="Date pickers"
-              metadata="MUI X"
-              versionSelector={renderVersionSelector([
-                // DATE_PICKERS_VERSION is set from the X repo
-                {
-                  ...(process.env.DATE_PICKERS_VERSION.startsWith('6')
-                    ? {
-                        text: `v${process.env.DATE_PICKERS_VERSION}`,
-                        current: true,
-                      }
-                    : {
-                        text: `v6-next`,
-                        href: `https://next.mui.com${languagePrefix}/components/data-grid/`,
-                      }),
-                },
-                {
-                  ...(process.env.DATE_PICKERS_VERSION.startsWith('5')
-                    ? {
-                        text: `v${process.env.DATE_PICKERS_VERSION}`,
-                        current: true,
-                      }
-                    : {
-                        text: `v5`,
-                        href: `https://mui.com${languagePrefix}/components/data-grid/`,
-                      }),
-                },
-              ])}
-            />
-          )}
+          <XProductIdentifier
+            pathname={canonicalAs}
+            languagePrefix={languagePrefix}
+            renderVersionSelector={renderVersionSelector}
+          />
           {canonicalAs.startsWith('/toolpad/') && (
             <ProductIdentifier name="Toolpad" metadata="MUI Toolpad" />
           )}
```